crates/oxc_transformer/src/commonjs/mod.rs

```diff
@@ -0,0 +1,138 @@
+                            ImportDeclarationSpecifier::ImportNamespaceSpecifier(ns) => {
+                                ns.local.name.as_str()
+                            }
+                            _ => unreachable!(),
+                        },
+                        &self.ctx.ast,
+                        false,
+                    )
+                } else {
+                    let assignees: Vec<(PropertyKey<'a>, BindingPattern<'a>)> = specifiers
+                        .iter()
+                        .map(|specifier| match specifier {
+                            ImportDeclarationSpecifier::ImportDefaultSpecifier(decl) => (
+                                self.ctx.ast.property_key_identifier_name(SPAN, "default"),
+                                self.ctx.ast.binding_pattern(
+                                    self.ctx.ast.binding_pattern_kind_binding_identifier(
+                                        SPAN,
+                                        &decl.local.name,
+                                    ),
+                                    None::<TSTypeAnnotation>,
+                                    false,
+                                ),
+                            ),
+                            ImportDeclarationSpecifier::ImportSpecifier(decl) => (
+                                match &decl.imported {
+                                    ModuleExportName::IdentifierName(name) => self
+                                        .ctx
+                                        .ast
+                                        .property_key_identifier_name(SPAN, name.name.as_str()),
+                                    ModuleExportName::StringLiteral(literal) => {
+                                        self.ctx.ast.property_key_expression(
+                                            self.ctx
+                                                .ast
+                                                .expression_string_literal(SPAN, &literal.value),
+                                        )
+                                    }
+                                    ModuleExportName::IdentifierReference(_) => unreachable!(),
+                                },
+                                self.ctx.ast.binding_pattern(
+                                    self.ctx.ast.binding_pattern_kind_binding_identifier(
+                                        SPAN,
+                                        &decl.local.name,
+                                    ),
+                                    None::<TSTypeAnnotation>,
+                                    false,
+                                ),
+                            ),
+                            ImportDeclarationSpecifier::ImportNamespaceSpecifier(_) => {
+                                unreachable!()
+                            }
+                        })
+                        .collect();
+                    import::create_named_require(
```

Comment:
Here import name declarations should reuse the identifier created by `create_namespaced_require`. Like 
``` .ts
import { a } from 'a'
import * as aa from 'a'
console.log(aa,a)
```
it should be transform to 
``` .ts
var _a = require('a')
console.log(_a,_a.a)
```